diff --git a/src/widget.c b/src/widget.c
index c87be63..4f0d11b 100644
--- a/src/widget.c
+++ b/src/widget.c
@@ -170,7 +170,7 @@ static void indicate_connectivity_internal(void) {
     };
 
 #if !IS_ENABLED(CONFIG_ZMK_SPLIT) || IS_ENABLED(CONFIG_ZMK_SPLIT_ROLE_CENTRAL)
-    struct zmk_endpoint_instance endpoint = zmk_endpoint_get_selected();
+    struct zmk_endpoint_instance endpoint = zmk_endpoints_selected();
     switch (endpoint.transport) {
     case ZMK_TRANSPORT_USB:
