diff --git a/app/src/behaviors/behavior_key_toggle.c b/app/src/behaviors/behavior_key_toggle.c
index ae06b14..e9ff699 100644
--- a/app/src/behaviors/behavior_key_toggle.c
+++ b/app/src/behaviors/behavior_key_toggle.c
@@ -38,10 +38,11 @@ static int on_keymap_binding_pressed(struct zmk_behavior_binding *binding,
     case OFF:
         return raise_zmk_keycode_state_changed_from_encoded(binding->param1, false,
                                                             event.timestamp);
-    case FLIP:
+    case FLIP: {
         bool pressed = zmk_hid_is_pressed(binding->param1);
         return raise_zmk_keycode_state_changed_from_encoded(binding->param1, !pressed,
                                                             event.timestamp);
+    }
     default:
         return -ENOTSUP;
     };
